src/NuGet.Core/NuGet.Packaging/PackageExtraction/StreamExtensions.cs

```diff
@@ -27,12 +34,24 @@ public static string CopyToFile(this Stream inputStream, string fileFullPath)
                 return fileFullPath;
             }
 
+            // For files of a certain size, we can do some Cleverness and mmap them instead
+            // of writing directly to disk. This can increase performance by a LOT, at the
+            // cost of files taking longer to get to disk -- but that's fine!
+            if (size > 0 && size <= MAX_MMAP_SIZE)
+            {
+                using (MemoryMappedFile mmf = MemoryMappedFile.CreateFromFile(fileFullPath, FileMode.Create, null, (long)size))
+                {
+                    MemoryMappedViewStream mmstream = mmf.CreateViewStream();
```

Comment:
You'll want to dispose the ViewStream stream to avoid leaking the handle, which will end up being cleaned up beyond the scope of this method. This might leave the MMF handle open.

Reply:
aaaaa yes thank you. This definitely seems to have been it, too!